src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -849,7 +850,292 @@ def coefficient_matrix(self, sparse=True):
+        #     row_indices: we need them for deg = target_degree - self[i].degree(),
+        #                  for all i = 0 ... len(self)-1
+        #     column_indices: we need them for deg = target_degree
+        # non homogeneous:
+        #     row_indices: we need them for deg = 0 ... target_degree - self[i].degree(),
+        #                  for all i = 0 ... len(self)-1,
+        #                  -> i.e. for deg = 0 ... target_degree - min(self[i].degree() for all i)
+        #     column_indices: we need them for deg = 0 ... target_degree
+        R_monomials_of_degree = {}
+        S_monomials_of_degree = {}
+        if homogeneous:
+            for poly_deg in set([poly.degree() for poly in self]):
+                deg = target_degree - poly_deg
+                R_monomials_of_degree[deg] = R.monomials_of_degree(deg)
+            S_monomials_of_degree[target_degree] = S.monomials_of_degree(target_degree)
+        else:
+            max_deg = target_degree - min(poly.degree() for poly in self)
+            for deg in range(max_deg + 1):
+                R_monomials_of_degree[deg] = R.monomials_of_degree(deg)
+            for deg in range(target_degree + 1):
+                S_monomials_of_degree[deg] = S.monomials_of_degree(deg)
+
+        # compute list of extended monomials (ring monomials + polynomial position)
+        # that will be used to construct the rows
+        row_indices = []
+        if homogeneous:
+            # order the rows with POT (or None)
+            if row_order is None or row_order == "TOP":
+                for i in range(m):
+                    deg = target_degree - self[i].degree()
+                    R_monomials_of_degree[deg].sort()
+                    row_indices += [(mon, i) for mon in R_monomials_of_degree[deg]]
+            # order the rows with TOP
+            else:
+                R_monomials_of_degree[deg].sort()
+                for mon in R_monomials_of_degree[deg]:
+                    row_indices += [(mon, i) for i in range(m)]
+        else:
+            # order the row with POT (or None)
+            if row_order is None or row_order == "TOP":
+                for i in range(m):
+                    R_monomials_usefull = []
+                    for deg in range(target_degree - self[i].degree() + 1):
+                        R_monomials_usefull += R_monomials_of_degree[deg]
+                    R_monomials_usefull.sort()
+                    row_indices += [(mon, i) for mon in R_monomials_usefull]
+            # order the row with TOP
+            else:
+                R_monomials_usefull = []
+                for deg in range(max_deg + 1):
+                    R_monomials_usefull += R_monomials_of_degree[deg]
+                R_monomials_usefull.sort()
+                for mon in R_monomials_usefull:
+                    row_indices += [(mon, i) for i in range(m)
+                                    if mon.degree() <= target_degree - self[i].degree() + 1]
```

Comment:
```suggestion
                                    if mon.degree() + self[i].degree() <= target_degree]
```

(there was a ``+1`` in the ``range`` of the previous version, but here with a non-strict inequality I think the ``+1`` should disappear)